server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/test/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutesTest.java

```diff
@@ -1590,6 +1680,171 @@ void restoreShouldNotAppendMessagesToAnOtherUserMailbox() throws Exception {
+                    .body("startedDate", is(notNullValue()))
+                    .body("submitDate", is(notNullValue()))
+                    .body("completedDate", is(notNullValue()));
+            }
+        }
+
+        @Nested
+        class SharingTest {
+
+            @Test
+            void exportShouldCallBlobSharingTargetToSharee() throws Exception {
+                vault.append(USER, DELETED_MESSAGE, new ByteArrayInputStream(CONTENT)).block();
+                vault.append(USER, DELETED_MESSAGE_2, new ByteArrayInputStream(CONTENT)).block();
+
+                String taskId =
+                    with()
+                        .queryParam("action", "export")
+                        .queryParam("sharee", USER_2.asString())
+                        .body(MATCH_ALL_QUERY)
+                        .post(USER.asString())
+                    .jsonPath()
+                        .get("taskId");
+
+                with()
+                    .basePath(TasksRoutes.BASE)
+                    .get(taskId + "/await");
+
+                verify(blobSharing, times(1))
+                    .share(eq(USER_2.asMailAddress()), any());
+            }
+
+            @Test
+            void exportShouldNotDeleteMessagesInTheVault() {
+                vault.append(USER, DELETED_MESSAGE, new ByteArrayInputStream(CONTENT)).block();
+                vault.append(USER, DELETED_MESSAGE_2, new ByteArrayInputStream(CONTENT)).block();
+
+                String taskId =
+                    with()
+                        .queryParam("action", "restore")
+                        .body(MATCH_ALL_QUERY)
+                        .post(USER.asString())
+                    .jsonPath()
+                        .get("taskId");
+
+                with()
+                    .basePath(TasksRoutes.BASE)
+                    .get(taskId + "/await");
+
+                assertThat(Flux.from(vault.search(USER, Query.ALL)).toStream())
+                    .containsOnly(DELETED_MESSAGE, DELETED_MESSAGE_2);
+            }
+
+            @Test
+            void exportShouldSaveDeletedMessagesDataToBlobStore() throws Exception {
+                vault.append(USER, DELETED_MESSAGE, new ByteArrayInputStream(CONTENT)).block();
+                vault.append(USER, DELETED_MESSAGE_2, new ByteArrayInputStream(CONTENT)).block();
+
+                String taskId =
+                    with()
+                        .queryParam("action", "export")
+                        .queryParam("sharee", USER_2.asString())
+                        .body(MATCH_ALL_QUERY)
+                        .post(USER.asString())
+                    .jsonPath()
+                        .get("taskId");
+
+                with()
+                    .basePath(TasksRoutes.BASE)
+                    .get(taskId + "/await");
+
+                byte[] expectedZippedData = zippedMessagesData();
+
+                assertThat(blobStore.read(blobIdFactory.forPayload(expectedZippedData)))
```

Comment:
To assert the content of deleted message inside blobstore. I need to retrieve corresponding `blodId`. There are 3 ways to do it:
- Currently, by reverse calculating the `blobId` from byte[] data, I expose to the test that I am using hasing `blobIdFactory`.
- Capturing `blobId` from `blobStore.save()` by Mockito.
- expose more content of MemoryBlobStore by adding methods in it. Then I expose to the test that I am using MemoryBlobStore.

There is a open question for which way should I go, with this test. Please leave your opinion!

Reply:
I'm OK with both the hash computation & adding extra methods to MemoryBlobStore